src/app/services/MusicPlayerApi.js

```diff
@@ -47,14 +47,58 @@ export async function shufflePlayPlaylist(playlist) {
   await music.player.play();
 }
 
-export async function playNext(track) {
-  await MusicKit.getInstance().player.queue.prepend({
-    items: [createMediaItem(track)],
-  });
+export async function playNext(item) {
+  let items;
+  const music = MusicKit.getInstance();
+  if (item.type === 'songs' || item.type === 'library-songs') {
+    items = [item];
+  } else if (item.type === 'albums' || item.type === 'library-albums') {
+    let albumData = item;
+    if (!albumData.relationships || !albumData.relationships.tracks) {
```

Comment:
Here, lines 57-60 for example. It could benefit from a util function to fetch a complete album if not already loaded.